mlir/utils/performance/quickTunerGen.py

```diff
@@ -0,0 +1,1115 @@
+
+    perf_configs.columns = perf_config_cols
+
+    perf_configs.drop(['param8', 'param9'], axis=1, inplace=True)
+
+    perf_configs['performance'] = final_df['NormalizedTFlops']
+
+    perf_configs = perf_configs.join(final_df[target_cols + ['DataType']])
+
+    grouped = {dtype[0]: df.drop('DataType', axis=1) for dtype, df in perf_configs.groupby(['DataType'])}
+
+    for k in grouped:
+        group = {cols: df.drop(target_cols, axis=1) for cols, df in grouped[k].groupby(target_cols)}
+        grouped[k] = group
+        
+    return grouped
+    
+
+def convertToConfig(type_df, filename, suffix=".qt", debug=False):
+    """
+    Converts the list of quickTuning sets into a group of files
+    """
+    if debug:
+        print(filename + suffix)
+        printConfigDict(type_df)
+    
+    for t in type_df:
+        fname = filename + "." + t + suffix
+        df = type_df[t]
+        if 'performance' in df.columns:
+            df = df.drop(labels=['performance'], axis=1)
+        df['forceUnroll'] = 1
+        df = df.to_csv(fname, index=False)
+
+"""
+Hardcoded tuner method
+"""
+
+class hardcodeQuickTune(quickTunerMethod):
+    """
+    Default quick tune method, uses preset values for the config 
+    file as of 07/22/24 these are the values that are coded into
+    GridwiseGemmParams.cpp
+    """
+    def __init__(self, name=None):
+        super().__init__(name)
+        self.default_f32 = pd.DataFrame({
+            "M/block": [256, 256, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 32, 32, 32, 32, 32, 32, 32, 32, 16, 16, 16, 16],
+            "N/block": [256, 64, 128, 128, 128, 64, 64, 64, 64, 64, 32, 16, 256, 128, 128, 128, 128, 64, 64, 64, 64, 64, 64, 64, 32, 32, 32, 16, 128, 128, 64, 64, 32, 32, 16, 16, 32, 32, 16, 16],
+            "K/block": [2, 8, 8, 4, 2, 8, 8, 8, 4, 2, 4, 4, 8, 4, 4, 4, 2, 8, 8, 8, 8, 4, 4, 8, 4, 8, 8, 8, 8, 8, 8, 4, 8, 8, 8, 4, 4, 8, 4, 8],
+            "M/wave": [128, 128, 64, 128, 32, 64, 32, 32, 32, 128, 128, 32, 64, 64, 64, 32, 32, 32, 16, 32, 16, 32, 16, 64, 32, 16, 16, 16, 32, 16, 32, 32, 16, 16, 16, 16, 16, 16, 16, 16],
+            "N/wave": [32, 32, 16, 32, 32, 16, 32, 16, 32, 32, 16, 16, 16, 32, 16, 16, 32, 32, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16, 16],
+            "kPack": [4, 1, 4, 4, 8, 1, 4, 1, 4, 4, 4, 8, 4, 1, 4, 4, 8, 4, 4, 4, 8, 4, 8, 8, 8, 4, 4, 8, 1, 4, 4, 4, 8, 4, 8, 8, 4, 8, 4, 8],
+            "forceUnroll": [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+        })
+
+        
+        self.default_f16 = pd.DataFrame({
+            "M/block": [128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 128, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 64, 32, 32, 32, 32, 32, 32, 16, 16, 16, 16],
```

Comment:
.. what's this about? Why do we have this? Where are the notes to keep it up to date?

Reply:
Also, if you're doing to do this, build the dataframe from a list of lists and give the column names separately - this is unreadable.